### PR TITLE
TMEDIA-585 - Add generator for content source block

### PR DESCRIPTION
**Pull request by matthewroach**
## Description
Add a generator for content source blocks to boost themes developer productivity and provide a consistent way to create a starting point for a new content source block with all the boilerplate code and files

## Jira Ticket
- [TMEDIA-585](https://arcpublishing.atlassian.net/browse/TMEDIA-585)

## Test Steps

1. Checkout this branch `git checkout TMEDIA-584-content-source-generator`
2. Run `npm run generate:content-source`
3. Enter a name for your content source - `commerce`
4. Verify the correct files where created for the new content source block named "Commerce Content Source Block"
5. In feature pack repo add the following to the block array in the block.json file - `"@wpmedia/commerce-content-source-block@local",`
6. Start fusion `npx fusion start -f -l @wpmedia/commerce-content-source-block`
7. Verify the content source shows up in the debugger - http://localhost/pagebuilder/tools/debugger?_website=the-gazette&type=content-debugger

Note - Content source will not return any data in debugger as it's not hitting an Arc API endpoint

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x] Confirmed this PR has unit test files
  - [x] Ran `npm run test`, made sure all tests are passing
  - [x] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: _templates/content-source/new/prompt.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = [
+  {
+    type: 'input',
+    name: 'block_name',
+    message: 'What is the content source block called? Type `collection` for Collection Source Block.',
```
Comment (JackHowa) [inline _templates/content-source/new/prompt.js]: would this be Collection Content Source Block? looking at the package.json
```suggestion
    message: 'What is the content source block called? Type `collection` for Collection Content Source Block.',
```

File: _templates/content-source/new/readme.ejs.t [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+_Please provide a 1-2 sentence description of what the @wpmedia/<%= h.inflection.dasherize(block_name) %>-content-source-block is and what it does._
+
+## ANS Schema
+Outline any schema information requirements necessary to know for ths block
+
+## Configurable Params
+None
+
+## TTL
+- `120`
```
Comment (JackHowa) [inline _templates/content-source/new/readme.ejs.t:15]: nice

**Review by JackHowa — DISMISSED** [review]
Looks good! Nice work with the documentation and defaults. 

Noticed one small semantic change in the prompt. And can you add an npmignore for the test and mock files? I know for content sources they can get big. In the chain generator, there's an example npmignore

```
npm notice === Tarball Contents === 
npm notice 225B sources/collection-news.js     
npm notice 564B sources/collection-news.test.js
npm notice 555B package.json                   
npm notice 450B README.md                      
npm notice === Tarball Details === 
npm notice name:          @wpmedia/collection-news-content-source-block
npm notice version:       0.1.0                                        
npm notice package size:  1.0 kB                                       
npm notice unpacked size: 1.8 kB                                       
npm notice shasum:        1dc01790f4ea72b065b11d9422589b5e962441da     
npm notice integrity:     sha512-QnHmprf1ZxgP2[...]kQ7Tcy/mM4/Fw==     
npm notice total files:   4                                            
npm notice 
+ @wpmedia/collection-news-content-source-block@0.1.0
```

**Comment by matthewroach** [comment]
@JackHowa Made updates based on feedback

**Review by JackHowa — APPROVED** [review]
nice work on this

**Pull request merged**